tests/unit/errors.spec.ts

```diff
@@ -22,6 +22,14 @@ describe('Errors: ', () => {
         expect(custom.is).toHaveProperty('NoSuchEntity', true);
     });
 
+    it('should allow additional metadata fields', () => {
+        const error = errors.InvalidArgument;
+        const custom = error.addMetadataEntry('fieldName', [{ key: 'value' }]);
+        expect(custom.metadata.get('fieldName')).toBeTruthy();
+        expect(custom.metadata.get('fieldName')).toEqual([{ key: 'value' }]);
+        expect(custom.is).toHaveProperty('InvalidArgument', true);
+    });
+
```

Comment:
We probably should check that `error` is not modified also.

Reply:
To be more precise, I would add something like `expect(error.metadata.get('fieldName')).toBeNull()`.